central/detection/lifecycle/manager_impl.go

```diff
@@ -289,15 +289,21 @@ func (m *managerImpl) addToIndicatorQueue(indicator *storage.ProcessIndicator) {
 func (m *managerImpl) addBaseline(deploymentID string) []*storage.ProcessBaseline {
 	defer centralMetrics.SetFunctionSegmentDuration(time.Now(), "AddBaseline")
 
-	// Simply use search to find the process indicators for the deployment
-	indicatorSlice, _ := m.processesDataStore.SearchRawProcessIndicators(
-		lifecycleMgrCtx,
-		search.NewQueryBuilder().
-			AddExactMatches(search.DeploymentID, deploymentID).
-			ProtoQuery(),
-	)
-
-	return m.buildMapAndCheckBaseline(indicatorSlice)
+	// Group the processes into particular baseline segments
+	baselineMap := make(map[processBaselineKey][]*storage.ProcessIndicator)
+
+	fn := func(indicator *storage.ProcessIndicator) error {
+		key := indicatorToBaselineKey(indicator)
+		baselineMap[key] = append(baselineMap[key], indicator)
+		return nil
+	}
+
+	query := search.NewQueryBuilder().
+		AddExactMatches(search.DeploymentID, deploymentID).
+		ProtoQuery()
+	_ = m.processesDataStore.GetByQueryFn(lifecycleMgrCtx, query, fn)
```

Comment:
Interesting we were squashing the error before.  We should probably log it at least.